### PR TITLE
feat(expandable): add expandable

### LINKED ISSUE #262: Expansion panel component
Are there any immediate plans to implement an expansion panel similar to https://material.angular.io/components/expansion/examples or https://aurelia-ui-toolkits.github.io/demo-materialize/#/samples/collapsible/basic-use ?

### PR DESCRIPTION
**Pull request by MaximBalaganskiy**
Close #262

### CODE REVIEW

File: packages/components/src/index.ts [hunk 1]
```diff
@@ -1,35 +1,37 @@
-import { FrameworkConfiguration, PLATFORM } from 'aurelia-framework';
```
Comment (bigopon) [inline packages/components/src/index.ts:1]: Shouldn't this file not affected by the line ending change any more? Can you help merge master into this branch

File: packages/expandable/src/ux-expandable.css.map [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{
```
Comment (bigopon) [inline packages/expandable/src/ux-expandable.css.map:1]: Should we not include this?

File: packages/expandable/src/ux-expandable.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  openChanged() {
+    this.openBoolean = normalizeBooleanAttribute('open', this.open);
+    this.updateContainerHeight();
+    this.element.dispatchEvent(new CustomEvent(UxExpandable.OPEN_CHANGED, { detail: this.openBoolean }));
+  }
+
+  @bindable
+  accordion: boolean | string = false;
+
+  setContentContainerHeightToAuto = () => {
```
Comment (bigopon) [inline packages/expandable/src/ux-expandable.ts:39]: should use `handleEvent` instead:
```ts
handleEvent(e) {
  switch (e.type) {
    case ('transitionend'):
      this.setContentContainerHeightToAuto();
      break;
  }
}

...
this.contentContainer.addEventListener('transitionend', this);
```

File: packages/expandable/src/ux-expandable.ts [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+import { UxComponent, StyleEngine, normalizeBooleanAttribute } from '@aurelia-ux/core';
+import { UxExpandableTheme } from './ux-expandable-theme';
+
+@inject(Element, StyleEngine, TaskQueue)
+@customElement('ux-expandable')
+@useView(PLATFORM.moduleName('./ux-expandable.html'))
+export class UxExpandable implements UxComponent {
+  constructor(public element: HTMLElement, private styleEngine: StyleEngine, private taskQueue: TaskQueue) { }
+
+  static OPEN_CHANGED = 'open-changed';
```
Comment (bigopon) [inline packages/expandable/src/ux-expandable.ts:11]: the naming of this should indicate what it is better. We can either:
- export an enum for all event: 
```ts
export const enum UxExpandableEvent {
  OpenChange = 'open-changed';
}
```

or change this name to `OPEN_CHANGED_EVENT`. Though I think a separate event export could probably be better. Not 100% sure on this

File: packages/expandable/src/ux-expandable.ts [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+    this.styleEngine.applyTheme(newValue, this.element);
+  }
+
+  public openBoolean: boolean = false;
+  @bindable
+  open: boolean | string = false;
+  openChanged() {
+    this.openBoolean = normalizeBooleanAttribute('open', this.open);
+    this.updateContainerHeight();
+    this.element.dispatchEvent(new CustomEvent(UxExpandable.OPEN_CHANGED, { detail: this.openBoolean }));
```
Comment (bigopon) [inline packages/expandable/src/ux-expandable.ts:33]: if `detail` is simply a boolean, we may later run into a need for a breaking change. Let's do
```ts
detail: { open: this.openBoolean }
```
so if needed, we can add extra properties without any issues. Can also start doing:
```ts
detail { component: this, open: this.openBoolean }
```

File: packages/components/src/index.ts (hunk 1 above)
Comment (MaximBalaganskiy) [reply packages/components/src/index.ts]: eols fixed. I've merged the master yesterday after another PR got accepted

File: packages/expandable/src/ux-expandable.ts [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+      case 'transitionend':
+        this.setContentContainerHeightToAuto();
+        break;
+    }
+  }
+
+  setContentContainerHeightToAuto() {
+    this.content.style.overflow = "visible";
+    this.content.style.height = "auto";
+    this.content.removeEventListener("transitionend", this);
```
Comment (bigopon) [inline packages/expandable/src/ux-expandable.ts:50]: it doesn't seem `transitionend` event listener is added anywhere for this? Did you miss it in update container height?

**Comment by bigopon** [comment]
@MaximBalaganskiy it's maybe a bit unreasonable, but I remember you have a online demo, given that we don't have setup for online demo for PR, could you help with deploying the example app so we can have a play please 😛

**Comment by MaximBalaganskiy** [comment]
Demo updated with expandable, sidenav is gone though

File: packages/expandable/src/ux-expandable.ts [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+    this.styleEngine.applyTheme(newValue, this.element);
+  }
+
+  public openBoolean: boolean = false;
+  @bindable
+  open: boolean | string = false;
+  openChanged() {
+    this.openBoolean = normalizeBooleanAttribute('open', this.open);
+    this.updateContainerHeight();
+    this.element.dispatchEvent(new CustomEvent(UxExpandable.OPEN_CHANGED_EVENT, { detail: { component: this, open: this.openBoolean } }));
```
Comment (bigopon) [inline packages/expandable/src/ux-expandable.ts:33]: can this be in multiple lines? We don't have any rule yet but having a massive line like this is gonna be refactored anyway. Maybe?
```suggestion
    this.element.dispatchEvent(new CustomEvent(
      UxExpandable.OPEN_CHANGED_EVENT,
      { detail: { component: this, open: this.openBoolean } }
    ));
```

**Comment by bigopon** [comment]
Looks good to me 👍

**Pull request merged**